test/recipes.spec.js

```diff
@@ -333,4 +331,115 @@ describe('Recipes', () => {
         });
     });
   });
+
+  describe('Pagination', () => {
+    const url = '/api/recipes';
+    let items;
+    before(async () => {
+      const array = Array(50)
+        .fill()
+        .map(() => {
+          return chai
+            .request(server)
+            .post('/api/recipes')
+            .set({ authorization: runtimeFixture.token })
+            .send(recipe);
+        });
+      await Promise.all(array);
+    });
+
+    it('should get all recipes when no page or limit is set', done => {
+      chai
+        .request(server)
+        .get(url)
+        .end((err, res) => {
+          const { recipes } = res.body;
+          items = recipes;
+          expect(res).to.have.status(200);
+          expect(recipes)
+            .to.be.an('Array')
+            .and.to.have.lengthOf.at.least(50);
+          done(err);
+        });
+    });
+
+    it('should get the first 30 recipes when limit is 30 but page is not set', done => {
+      chai
+        .request(server)
+        .get(`${url}?limit=30`)
+        .end((err, res) => {
+          const { recipes } = res.body;
+          expect(res).to.have.status(200);
+          expect(recipes)
+            .to.be.an('Array')
+            .and.has.lengthOf(30);
+          expect(recipes[0].id).to.equal(items[0].id);
+          done(err);
+        });
+    });
+
+    it('should get the second set of 10 recipes when page is 2 and no limit is set', done => {
+      chai
+        .request(server)
+        .get(`${url}?page=2`)
+        .end((err, res) => {
+          const { recipes } = res.body;
+          expect(res).to.have.status(200);
+          expect(recipes).to.be.an('Array');
+          expect(recipes.length).to.equal(10);
+          expect(recipes[0].id).to.not.equal(items[0].id);
```

Comment:
You should test positives, not just negatives. (i.e. the fact that the first returned recipe is not the first created recipe does not guarantee whether the right recipes were returned or not). That applies to the rest of the examples after this here as well